src/content/en/fundamentals/media/fast-playback-with-video-preload.md

```diff
@@ -0,0 +1,569 @@
+  video.play().then(_ => {
+    // If preloaded video URL was already cached, playback started immediately.
+  });
+</script>
+```
+
+Note: I would recommend using this only for small media files (< 5MB).
+
+Because the preloaded resource is going to be consumed by a video element in
+the example, the `as` preload link value is `video`. If it were an audio
+element, it would be `as="audio"`.
+
+### Preload first segment
+
+The example below shows how to preload the first segment of a video with `<link
+rel="preload">` and use it with Media Source Extensions. If you're not familiar
+with the MSE Javascript API, please read [MSE basics].
+
+For the sake of simplicity, let's assume the entire video has been split into
+smaller files like "file_1.webm", "file_2.webm", "file_3.webm", etc.
+
+```
+<link rel="preload" as="fetch" href="https://cdn.com/file_1.webm">
+
+<video id="video" controls></video>
+
+<script>
+  const mediaSource = new MediaSource();
+  video.src = URL.createObjectURL(mediaSource);
+  mediaSource.addEventListener('sourceopen', sourceOpen, { once: true });
+
+  function sourceOpen() {
+    URL.revokeObjectURL(video.src);
+    const sourceBuffer = mediaSource.addSourceBuffer('video/webm; codecs="vp9"');
+
+    // If video is preloaded already, fetch will return immediately a response
+    // from the browser cache (memory cache). Otherwise, it will perform a
+    // regular network fetch.
+    fetch('https://cdn.com/file_1.webm')
+    .then(response => response.arrayBuffer())
+    .then(data => {
+      // Append the data into the new sourceBuffer.
+      sourceBuffer.appendBuffer(data);
+      // TODO: Fetch file_2.webm when user starts playing video.
+    })
+    .catch(error => {
+      // TODO: Show "Video is not available" message to user.
+    });
+  }
+</script>
+```
+
+Warning: For cross-origin resources, make sure your CORS headers are set
+properly. As we can't create an array buffer from an opaque response retrieved
+with `fetch(videoFileUrl, { mode: 'no-cors' })`, we won't be able to feed any
+video or audio element.
+
+### Support
+
+Link preload is not supported in every browser yet. You may want to detect its
+availability with the snippets below to adjust your performance metrics.
+
+```
+function preloadFullVideoSupported() {
+  const link = document.createElement('link');
+  link.as = 'video';
+  return (link.as === 'video');
+}
+
+function preloadFirstSegmentSupported() {
+  const link = document.createElement('link');
+  link.as = 'fetch';
+  return (link.as === 'fetch');
+}
+```
+
+## Manual Buffering
+
+Before we dive into the [Cache API] and service workers, let's see how to
+manually buffer a video with MSE. The example below assumes that your web
+server supports HTTP Range requests but this would be pretty similar with
+segment files. Note that some middleware libraries such as [Google's Shaka
+Player], [JW Player], and [Video.js] are dedicated to handle this for you.
```

Comment:
'dedicated to handle'

I think you mean 'built to handle'. If so, please change.